api/db/repository/petition.py

```diff
@@ -62,12 +84,74 @@ def get_by_id(_db: Session, petition_id: int, petition_status=""):
     return db_petition.first()
 
 
-def get_all(_db: Session, offeset: int = 0, limit: int = 100):
-    return _db.query(Petition).offset(offeset).limit(limit).all()
+def get_all(
+    _db: Session,
+    offset: int = 0,
+    limit: int = 100,
+    statuses: List[str] = None,
+    creation_date: List[str] = None,
+    due_date: List[str] = None,
+    ordering: List[str] = None,
+    search_query: str = "",
+):
+    query = _db.query(Petition).filter(
+        Petition.title.contains(search_query, autoescape=True)
+    )
+    if statuses:
+        query = query.filter(Petition.status.in_(statuses))
 
+    try:
+        if creation_date:
+            query = __form_date_order(
+                query, creation_date, "creation_date"
+            )
+        if due_date:
+            query = __form_date_order(query, due_date, "due_date")
+    except Exception as exc:
```

Comment:
general Exception are not recommended, what type of exception exactly are you expecting here?

Reply:
I changed it to `ValueError`.